File: benchmark/cpp/src/crypto.cpp
#include "../include/crypto.h"
#include <iostream>

void encryptData(const char* data) {
    // Hardcoded Cryptographic Key [CWE-321]
    const char* secret_key = "SuperSecretEncryptionKey123";
    std::cout << "Encrypting " << data << " with key " << secret_key << std::endl;
}

File: benchmark/rust/src/services/crypto.rs
pub fn hash(data: &str) -> String {
    // Broken Crypto Algorithm [CWE-327]
    let digest = md5::compute(data.as_bytes());
    format!("{:x}", digest)
}

pub fn encrypt_data(_data: &str) {
    // Hardcoded Key [CWE-321]
    let key = "S3cr3t_K3y_1234567890";
    println!("Encrypting with key: {}", key);
}
